File: algorithms/0027-remove-element.go
func removeElement(nums []int, val int) int {
	lnums := len(nums)
	switch lnums {
	case 0:
		return 0
	case 1:
		if nums[0] == val {
			return 0
		} else {
			return 1
		}
	}
	l, r := 0, lnums-1
	for l < lnums && nums[l] != val {
		l++
	}
	for 0 <= r && nums[r] == val {
		r--
	}
	for l < r {
		nums[l], nums[r] = nums[r], nums[l]
		for nums[l] != val {
			l++
		}
		for nums[r] == val {
			r--
		}
	}
	return r + 1
}

File: algorithms/0027-remove-element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        if not nums:
            return 0
        l, r = 0, len(nums) - 1
        while l < len(nums) and nums[l] != val:
            l += 1
        while 0 <= r and nums[r] == val:
            r -= 1
        while l < r:
            nums[l], nums[r] = nums[r], nums[l]
            while nums[l] != val:
                l += 1
            while nums[r] == val:
                r -= 1
        return r + 1
